### DISCUSSION TITLE
How to create ContextMenu items newly on every request?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [AvaloniaUI/Avalonia#15779] [Q&A] [answered]
I want to create a `ContextMenu` with items are created on the fly and may change for each opening of the `ContextMenu`. I encountered the situation that the `ContextMenu` is sampled at the first time it is requested. It does not update afterwards.

How can I create a `ContextMenu` with items that are newly created on every request?

**Comment by maxkatz6** [comment, marked as answer]
You can handle ContextRequested event.

**Reply by DevAM-Tools** [reply]
Thanks for the hint. It is important detail that `ContextRequested` only seems to be triggered if no `ContextMenu` is defined in XAML.
